server/clients/eth/erc20.go

```diff
@@ -1,60 +1,102 @@
 package goethclient
 
 import (
+	"context"
 	"math/big"
 
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
 	"github.com/threefoldtech/web3_proxy/server/clients/eth/erc20"
 )
 
-func (c *Client) GetTokenBalance(contractAddress, target string) (*big.Int, error) {
+func (c *Client) GetTokenBalance(contractAddress string) (*big.Int, error) {
 	token, err := erc20.NewErc20(common.HexToAddress(contractAddress), c.Eth)
 	if err != nil {
 		return nil, err
 	}
 
-	return token.BalanceOf(&bind.CallOpts{}, common.HexToAddress(target))
+	return token.BalanceOf(&bind.CallOpts{}, c.Address)
 }
 
-func (c *Client) TransferTokens(contractAddress, target string, amount int64) (string, error) {
+func (c *Client) TransferTokens(ctx context.Context, contractAddress, target string, amount int64) (string, error) {
```

Comment:
There is a new way of getting the state from the session. See stellar:
```
// Transer an amount of TFT from the loaded account to the destination.
func (c *Client) Transfer(ctx context.Context, conState jsonrpc.State, args Transfer) error {
	state := State(conState)
	if state.Client == nil {
		return pkg.ErrClientNotConnected{}
	}

	return state.Client.Transfer(args.Destination, args.Memo, args.Amount)
}
```
Maybe we should use that? I believe that is the way to go from now on.